examples/query-filters/queryFilters.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+const Fastify = require('fastify')
+const GQL = require('../..')
+const app = Fastify()
+const sqlite = require('sqlite')
+
+const dbPromise = sqlite.open('./database.sqlite', { promise: Promise })
+  .then(db => db.migrate({ force: 'last' })).catch((err) => {
+    console.log(err)
+  })
+
+const schema = `
+  type Post {
+    id: ID!
+    title: String!
+    subtitle: String
+    content: String
+    category: Category
+  }
+
+  type Category {
+    id: ID!
+    name: String!
+  }
+
+  type Query {
+    posts(limit: Int): [Post]
+  }
+`
+
+const resolvers = {
+  Query: {
+    posts: async (_, params, context, info) => {
+      const queryData = context.buildQueryObject(info)
+      const db = await dbPromise
+      let posts
+      let relation = 'category'
+
+      if (queryData.hasRelation(relation)) {
+        // Select with relation
+        posts = await db.all(
+          `SELECT  ${queryData.getRootFields()}, ${queryData.getRelationFields(relation)} 
+          FROM Post INNER JOIN Category ON Category.id = Post.categoryId`)
```

Comment:
Would you mind adding a comment that this can have a SQL injection?

Reply:
Sorry, that took me long to get back on this. I was actually thinking about how to do that properly to avoid a couple of problems. Using low-level SQL will not only mean SQL injection but also compatibility issues ( for example select table.field __on__ table__field will require __AS__ in PostgresDB). I think to do this properly we can:

- [ ] Ignore those helpers that were written and made them part of example which will mean that we deliver fields only and users will need to apply their own pattern depending on their ORM. It makes sense as there will be so many dialects to support.

- [ ] Use more opinionated and high-level query mechanism in helpers:
nearform/sql or even knex. This will mean that things will work out of the box for some of the common databases that are supported by this libs. This will mean that helpers will be more solid and production ready but the general pattern will be not that simple to understand and map to non supported libraries.

- [ ] Provide helpers like now but mark them as just examples and provide variations for different dialects. I spent some time cracking GraphQL specification to see if it is possible to pass field value as invalid js type name and it does not seem to be passing validation.